File: 【队列】2023C-篮球游戏.cpp
/*
 * @hw id=2023C lang=cpp
 *
 * 篮球游戏
 */

// @hw code=start
#include<iostream>
#include<vector>
#include<sstream>
using namespace std;

int main () {
    string s, sn, token;
    // cin >> s >> sn;
    getline(cin, s);
    getline(cin, sn);
    vector<string> ss, sns, q;
    stringstream ss1(s);
    stringstream ss2(sn);

    while(getline(ss1, token, ',')) {
        ss.push_back(token);
    }
    while(getline(ss2, token, ',')) {
        sns.push_back(token);
    }

    int i = 0;
    string ans;
    for(string si : ss) {
        q.push_back(si);
        while(q.size() > 0) {
            if(q[0] == sns[i]) {
                ans += "L";
                q.erase(q.begin());
                i++;
            } else if(q[q.size() - 1] == sns[i]) {
                ans += "R";
                q.pop_back();
                i++;
            } else {
                break;
            }
        }
    }
    if(ans.length() == sns.size()) {
        cout << ans << endl;
    } else {
        cout << "NO" << endl;
    }
    return 0;
}
// @hw code=end

File: 【队列】2023C-篮球游戏.py
#
# @hw id=2023C lang=python3
#
# 篮球游戏
#

# @hw code=start
ss = input().split(",")
sns = input().split(",")
q = list()
i = 0
ans = str()

for s in ss:
    q.append(s)
    while(len(q) > 0):
        if q[0] == sns[i]:
            q = q[1:]
            i += 1
            ans += "L"
        elif q[-1] == sns[i]:
            q = q[:-1]
            i += 1
            ans += "R"
        else:
            break
if len(ans) == len(sns):
    print(ans)
else:
    print("NO")
# @hw code=end